data/helpers.d/system

```diff
@@ -53,3 +53,92 @@ ynh_abort_if_errors () {
 ynh_get_debian_release () {
 	echo $(lsb_release --codename --short)
 }
+
+# Start (or other actions) a service,  print a log in case of failure and optionnaly wait until the service is completely started
+#
+# usage: ynh_systemd_action [-n service_name] [-a action] [ [-l "line to match"] [-p log_path] [-t timeout] [-e length] ]
+# | arg: -n, --service_name= - Name of the service to reload. Default : $app
+# | arg: -a, --action=       - Action to perform with systemctl. Default: start
+# | arg: -l, --line_match=   - Line to match - The line to find in the log to attest the service have finished to boot.
+#                              If not defined it don't wait until the service is completely started.
+# | arg: -p, --log_path=     - Log file - Path to the log file. Default : /var/log/$app/$app.log
+# | arg: -t, --timeout=      - Timeout - The maximum time to wait before ending the watching. Default : 300 seconds.
+# | arg: -e, --length=       - Length of the error log : Default : 20
+ynh_systemd_action() {
+    # Declare an array to define the options of this helper.
+    declare -Ar args_array=( [n]=service_name= [a]=action= [l]=line_match= [p]=log_path= [t]=timeout= [e]=length= )
+    local service_name
+    local action
+    local line_match
+    local length
+    local log_path
+    local timeout
+
+    # Manage arguments with getopts
+    ynh_handle_getopts_args "$@"
+
+    local service_name="${service_name:-$app}"
+    local action=${action:-start}
+    local log_path="${log_path:-/var/log/$service_name/$service_name.log}"
+    local length=${length:-20}
+    local timeout=${timeout:-300}
+
+    # Start to read the log
+    if [[ -n "${line_match:-}" ]]
+    then
+        local templog="$(mktemp)"
+	# Following the starting of the app in its log
+	if [ "$log_path" == "systemd" ] ; then
+            # Read the systemd journal
+            journalctl -u $service_name -f --since=-45 > "$templog" &
```

Comment:
Considering my answer for tail, I'm wondering why there's a --since minus 45s here !?
@JimboJoe could you explain why you added that ? https://github.com/YunoHost-Apps/Experimental_helpers/pull/18/commits/97d6d5261c5745f361061beb64dd3a885215e287
Is `--lines=` not working the same way than for tail ?

Reply:
For me a -n0 -q (--lines=0 and --quiet) do exactly what we want.

Reply:
Is it `--quiet` which prevent the last previous line to be printed ?

Reply:
Retried right now, `-n0 -q` gives me immediately the last line of the log, even if nothing's happening in this log.
So I rather prefer `--since=0` which prints only following lines.

Reply:
Weird for `-n0` but ok, anyway, --since=0 is working